File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        buckets = [[] for _ in range(len(nums) + 1)]
        valueCounts = {}
        res = []
        for n in nums:
            if n in valueCounts:
                valueCounts[n] += 1
            else:
                valueCounts[n] = 1
        
        for num,count in valueCounts.items():
            buckets[count].append(num)
        
        for i in range(len(buckets) -1, -1, -1):
            for n in buckets[i]:
                res.append(n)
                if len(res) == k:
                    return res
        
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> counts;
        for(const int& n : nums){
            counts[n]++;
        }

        for(const auto [key, value] : counts){
            if(value >= 2) return true;
        }
        
        return false;
    }
};